src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -84,7 +83,7 @@ private static Metadata GetMetadataNoThrow(PortableExecutableReference reference
             var metadataId = GetMetadataIdNoThrow(reference);
             if (metadataId == null)
             {
-                return SpecializedTasks.Default<SymbolTreeInfo>();
+                return Task.FromResult(CreateEmpty(checksum));
```

Comment:
:bulb: If this is expected in production and the result doesn't get mutated, cache this `Task<SymbolTreeInfo>`.